compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -2681,8 +2681,17 @@ impl<'a> Parser<'a> {
         Ok((pat, expr))
     }
 
-    /// Parses `for <src_pat> in <src_expr> <src_loop_block>` (`for` token already eaten).
+    /// Parses `for await? <src_pat> in <src_expr> <src_loop_block>` (`for` token already eaten).
     fn parse_expr_for(&mut self, opt_label: Option<Label>, lo: Span) -> PResult<'a, P<Expr>> {
+        let is_await =
+            self.token.uninterpolated_span().at_least_rust_2018() && self.eat_keyword(kw::Await);
+
+        if is_await {
+            self.sess.gated_spans.gate(sym::async_for_loop, self.prev_token.span);
+        }
+
+        let kind = if is_await { ForLoopKind::ForAwait } else { ForLoopKind::For };
```

Comment:
Is there any reason in particular to not write this as follow with a single `if` check?

```suggestion
        let is_await =
            self.token.uninterpolated_span().at_least_rust_2018() && self.eat_keyword(kw::Await);
        let kind = if is_await {
            self.sess.gated_spans.gate(sym::async_for_loop, self.prev_token.span);
            ForLoopKind::ForAwait
        } else {
            ForLoopKind::For
        };
```